hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystemStore.java

```diff
@@ -521,12 +523,14 @@ public OutputStream createFile(final Path path,
             isNamespaceEnabled ? getOctalNotation(permission) : null,
             isNamespaceEnabled ? getOctalNotation(umask) : null,
             isAppendBlob,
-            null);
+            null,
+            lease);
       }
       perfInfo.registerResult(op.getResult()).registerSuccess(true);
 
-      AbfsLease lease = maybeCreateLease(relativePath);
-
+      if (lease == null) {
```

Comment:
Acquiring lease along with create/append enables atomic fetch of lease with the store operation and prevent any parallel writers. If configured for infinite lease, the path creation will happen without bundled lease acquiry and rely on later acquireLease API call in AbfsLease. 
AbfsLease background acquireLease should be scheduled post lease is acquired along with create API (bundled) ? Infinite lease flow purely on acquireLease API should be the case only if fs.azure.write.enforceLease is off.